docs/guides/es2k/es2k-guide.md

```diff
@@ -0,0 +1,212 @@
+Note: max vport you can pass here from [0-6] "eal-args": "--lcores=1-2 -a <cpf_bdf>,vport=[0-1] -- -i --rxq=1 --txq=1 --hairpinq=1 --hairpin-mode=0x0"
+
+Eg. "eal-args": "--lcores=1-2 -a 00:01.6,vport=[0-1] -- -i --rxq=1 --txq=1 --hairpinq=1 --hairpin-mode=0x0"
+
+We give options to each process to request numbers of configure queues before starting the process
+
+Admin must set cfgqs-idx between 0-15. See the latest es2k_skip_p4.conf for the exact syntax.
+If you want 16 cfgqs, use "cfgqs-idx": "0-15". Specify the range as a string. Index numbers from 0 to 15 are supported.
+In multi process environment, user should plan and split the queues between primary and secondary processes and specify
+the range with cfgqs-idx parameter. Total number of queues split between processes should not exceed 16.
+
+#### Run the infrap4d daemon
+
+```bash
+cd $IPDK_RECIPE
+sudo ./install/sbin/infrap4d
+```
+
+*Note*: By default, infrap4d runs in detached mode. If you want to run in
+attached mode, specify the --nodetach command-line option.
+
+- All infrap4d logs are by default logged under `/var/log/stratum`.
+- All P4SDE logs are logged in `p4_driver.log` under `$IPDK_RECIPE`.
+*-All OVS logs are logged under `/tmp/ovs-vswitchd.log`.
+
+### Run a sample program
+
+Open a new terminal to set the pipeline and try the sample P4 program.
+Set up the environment and export all environment variables to sudo user.
+
+```bash
+source ./scripts/es2k/setup_env.sh $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL
+./scripts/es2k/copy_config_files.sh $IPDK_RECIPE $SDE_INSTALL
+alias sudo='sudo PATH="$PATH" HOME="$HOME" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" SDE_INSTALL="$SDE_INSTALL"'
+```
+
+#### Create IPDF netdevs
+
+After installing ATE Kernel on HOST machine, install the following drivers
+to bind the network devices (netdevs) to the E2100 target.
+
+```bash
+modprobe auxiliary 
+modprobe idpf
+```
+
+#### Create P4 artifacts
+
+- P4 programs are available in the P4 SDE repository.
+
+- Obtain a copy of the P4 Compiler for ES2K and install it on your
+  development system.
+
+- Set the environment variable OUTPUT_DIR to the location where artifacts
+  should be generated and where p4 files are available
+
+```bash
+export OUTPUT_DIR=/root/p4-driver/mev_reference_p4_files/simple_l3_l4_pna
+```
+
+- Generate the artifacts using the p4c compiler installed in the previous step:
+
+```bash
+p4c-pna-xxp -I/usr/lib -I/usr/share/p4c/p4include -I/usr/share/p4c/idpf-lib \
+            $OUTPUT_DIR/simple_l3_l4_pna.p4 -o $OUTPUT_DIR/simple_l3_l4_pna.s \
+            --p4runtime-files $OUTPUT_DIR/simple_l3_l4_pna.p4info.txt \
+            --context $OUTPUT_DIR/simple_l3_l4_pna.context.json \
+            --bfrt $OUTPUT_DIR/simple_l3_l4_pna.bfrt.json
+```
+
+*Note*: The above commands will generate three files
+(`simple_l3_l4_pna.p4info.txt`, `simple_l3_l4_pna.bfrt.json`, and
+`simple_l3_l4_pna.context.json`).
+
+- Modify `simple_l3_l4_pna.conf` to specify the correct paths for
```

Comment:
will user be re-using es2k_skip_p4.conf or he should copy mev_sample.conf to simple_l3_l4_pna.conf and then modify the content accordingly?

Reply:
So es2k_skip_p4.conf comes with p4cp-nws rpm and not sde, p4c rpm. So, I had to add instructions of using mev_sample.conf because only that's available to user with rpms installed. SDE P4 files don't include conf file in every folder.